Foundation/Array/Bitmap.hs

```diff
@@ -87,10 +87,13 @@ instance C.Collection Bitmap where
     null = null
     length = length
     elem e = Data.List.elem e . toList
-    minimum = Data.List.minimum . toList . C.getNonEmpty -- TODO can shortcircuit all this massively
-    maximum = Data.List.maximum . toList . C.getNonEmpty -- TODO DITTO
-    all p = Data.List.all p . toList
-    any p = Data.List.any p . toList
+    minimum bm = let bm' = C.getNonEmpty bm
+                 in foldl' min (unsafeIndex bm' 0) bm'
+    maximum bm = let bm' = C.getNonEmpty bm
+                 in foldl' max (unsafeIndex bm' 0) bm'
+    all p      = foldl' (\acc x -> p x && acc) True
+    any p      = foldl' (\acc x -> p x || acc) False
```

Comment:
Hmm, it does seem to. Fair enough then - I guess it's worth adding an explicit test of this property somewhere.